src/ORM/Locator/TableLocator.php

```diff
@@ -299,4 +306,29 @@ public function remove($alias)
             $this->_fallbacked[$alias]
         );
     }
+
+    /**
+     * Returns a singleton instance of LocatorInterface implementation.
+     *
+     * @return \Cake\ORM\Locator\LocatorInterface
+     */
+    public static function getInstance()
```

Comment:
The singleton doesn't remove static class usage it just moves it. This class would now have a mixture of static and non-static state, which doesn't seem great.